drivers/hmis/app/graphql/mutations/delete_service_category.rb

```diff
@@ -0,0 +1,27 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Mutations
+  class DeleteServiceCategory < BaseMutation
+    argument :id, ID, required: true
+
+    field :service_category, Types::HmisSchema::ServiceCategory, null: true
+
+    def resolve(id:)
+      raise 'access denied' unless current_user.can_configure_data_collection?
+
+      service_category = Hmis::Hud::CustomServiceCategory.find_by(id: id)
+      raise HmisErrors::ApiError, 'Invalid service category ID' unless service_category
```

Comment:
This is fine, I also think it's okay to just do `Hmis::Hud::CustomServiceCategory.find(id)` which will raise its own exception if not found

Reply:
Thanks, this is much tidier!